build-tools/packages/build-tools/src/fluidBuild/tasks/taskFactory.ts

```diff
@@ -151,17 +167,15 @@ export class TaskFactory {
 			return new GroupTask(node, command, context, subTasks, taskName);
 		}
 
-		// Resolve "npm run" to the actual script
-		if (command.startsWith("npm run ")) {
-			const scriptName = command.substring("npm run ".length);
-			const subTask = node.getScriptTask(scriptName, pendingInitDep);
-			if (subTask === undefined) {
-				throw new Error(
-					`${node.pkg.nameColored}: Unable to find script '${scriptName}' in 'npm run' command`,
-				);
+		// Resolve "npm run" (or other package manager's run command) to the actual script if possible
+		const runScript = getRunScriptName(command, node.pkg.packageManager);
+		if (runScript !== undefined) {
+			const subTask = node.getScriptTask(runScript, pendingInitDep);
+			if (subTask !== undefined) {
+				// Even though there is only one task, create a group task for the taskName
+				return new GroupTask(node, command, context, [subTask], taskName);
```

Comment:
This seems like a subtle logic change. Before we were erroring out when subtask is undefined, but now we'll fall through to the downstream logic. I don't understand why that's correct.

Also the "there is only one task" comment doesn't make sense to me. How do we know there's only one task? And which task is it referring to?

Reply:
Yes, the change is intentional.  Ability to parse `npm run` builds a better dependency graph, but without it, we can just treat `npm run` command as in any other commands (leaf task), and just execute it.   This is also support for the case where `npm run` add extra arguments (e.g. `npm run <script> -- <args>` or `pnpm run <script> <args>)

The ONLY task in this case is the <script> for `npm run <script>` command (which could be a group command or not), but we still create a "group task" for the current script name so that we can associate name with it.